<h1>Case Study on Measures of Central Tendency and Dispersion</h1>

<h3><b><i>An Institution wishes to find out their student’s ability in maths, reading and
writing skills. The Institution wants to do an exploratory study to check the
following information.</i></b></h3>

<b>
1. Find out how many males and females participated in the test.<br>
2. What do you think about the students' parental level of education?<br>
3. Who scores the most on average for math, reading and writing based on<br>
    ● Gender<br>
    ● Test preparation course<br>
4. What do you think about the scoring variation for math, reading and writing
based on<br>
    ● Gender<br>
    ● Test preparation course<br>
5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.<br>
</b>

<h2> Read the dataset to the python environment.</h2>

In [2]:
# Importing Pandas Library as 'pd'
import pandas as pd
# Importing NumPy Library as 'np'
import numpy as np

In [3]:
# Getting data from Local CSV File and copying into students_data Datafram
students_data=pd.read_csv('StudentsPerformance.csv')
# Getting data of dataframe students_data
students_data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

<h3><b>Checking null values</b></h3>

In [4]:
# Checking for null values
students_data[students_data.isna().any(axis=1)]

Empty DataFrame
Columns: [gender, race/ethnicity, parental level of education, lunch, test preparation course, math score, reading score, writing score]
Index: []

<b>### Insights:<br> 
    No null values found! </b>

<h2>1. Find out how many males and females participated in the test.</h2>

<b>Counting the number of female male customers.</b>

In [10]:
# To find the gender count using 'value_counts' method.
students_data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

<b>###Insights:<br>518 male and 482 female participants in students data.</b>

<h2>2. What do you think about the students' parental level of education?</h2>

In [6]:
# To find the parental level of education using 'value_counts' method.
students_data['parental level of education'].value_counts().to_frame()

parental level of education
some college                                226
associate's degree                          222
high school                                 196
some high school                            179
bachelor's degree                           118
master's degree                              59

<b>###Insights: <br>59 parents have master's degree, 118 parents have bachelor's degree, 222 parents have associate's degree, 226 parents has some college level education and 375 parents have high school level education.</b>

<h3>3. Who scores the most on average for math, reading and writing based on<br>
● Gender</h3>

In [7]:
#To find the average using 'mean' method.
(students_data[['gender','math score','reading score','writing score']].groupby('gender')).mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

<h4><b>###Insights: <br> Among these,<br> Males have most average math score as well as Females have most average reading and writing score.</b></h4>

<h3>● Test preparation course</h3>

In [22]:
#To find the average using 'mean' method.
(students_data[['test preparation course','math score','reading score','writing score']].groupby('test preparation course')).mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

<b>###Insights:<br>Among these,<br>The number of candidates who completed the test preparation course have the highest average math score, reading score and writing score.</b>

<h3><b>4. What do you think about the scoring variation for math, reading and writing based on<br>
● Gender</b></h3>

In [28]:
#To find the deviation using 'std' method.
(students_data[['gender','math score','reading score','writing score']].groupby('gender')).std()

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

<b>###Insights: <br> Among these,<br> The females has little high scoring variation than males on math, reading and writing score.</b>

<h3><b>● Test preparation course</b></h3>

In [26]:
#To find the deviation using 'std' method.
(students_data[['test preparation course','math score','reading score','writing score']].groupby('test preparation course')).std()

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

<b>###Insights:<br> Among these,<br> The candidate who could'nt compleated the test preparation cource has little higher scoring variation on math, reading and writing score than others.</b>

<h3><b>5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.</b></h3>

<h4><b> Summary stats on 'math score' column as frame.</b></h4>

In [32]:
#To get the summary of the column 'math score' using 'describe' method.
students_data["math score"].describe().to_frame()

math score
count  1000.000000
mean     66.001000
std      15.569567
min       0.000000
25%      57.000000
50%      66.000000
75%      77.000000
max     100.000000

<b>###Insights: <br> Among these, <br> The students those who are scored above 77.00 marks, they are eligible for the bonus points.</b>

In [6]:
#To get the values of  forth quartile using quartile().
q75 = students_data['math score'].quantile(q=0.75)
students_data[students_data['math score'].ge(q75)]

gender race/ethnicity parental level of education         lunch  \
414  female        group A                 high school  free/reduced   
415  female        group B             master's degree  free/reduced   
416  female        group B           bachelor's degree  free/reduced   
417  female        group B             master's degree      standard   
418  female        group B                 high school      standard   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
414               completed          77             88             85  
415               completed          77             97             94  
416                    none          77             85             87  
417                    none          77             90             84  
418               completed          77             82             89  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[255 rows x 8 columns]

<b>### Insights: <br> Among these, <br> This is basically values higher than the 75th quantile.Also the top 25% of values and 255 candidates get the bonus points.</b>